src/sagemaker/estimator.py

```diff
@@ -3846,8 +3848,20 @@ def _distribution_configuration(self, distribution):
             # smdistributed strategy selected
             if get_mp_parameters(distribution):
                 distribution_config["mp_parameters"] = get_mp_parameters(distribution)
+            # first make sure torch_distributed is enabled if instance type is p5
+            torch_distributed_enabled = distribution.get("torch_distributed").get("enabled", False)
             smdistributed = distribution["smdistributed"]
             smdataparallel_enabled = smdistributed.get("dataparallel", {}).get("enabled", False)
+            p5_enabled = True if self.instance_type == "p5.48xlarge" else False
+            img_uri = "" if self.image_uri == None else self.image_uri
```

Comment:
nit: Using `is None` is preferred than `== None`
- Try `pylint $filename` and it'll show you the suggestions

Reply:
Addressing this atm